compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1613,10 +1619,153 @@ public void visit(BLangSimpleVariable varNode) {
 
     @Override
     public void visit(BLangTupleVariable varNode) {
-        // assign the type to var type node
+        if (varNode.isDeclaredWithVar) {
+            // Will be handled in semantic analyzer.
+            return;
+        }
         if (varNode.type == null) {
             varNode.type = symResolver.resolveTypeNode(varNode.typeNode, env);
         }
+
+        // To support variable forward referencing we need to symbol enter each tuple member with type at SymbolEnter.
+        if (!(checkTypeAndVarCountConsistency(varNode, env))) {
+            varNode.type = symTable.semanticError;
+            return;
+        }
+    }
+
+    boolean checkTypeAndVarCountConsistency(BLangTupleVariable varNode, SymbolEnv env) {
+
+        Name varName = names.fromString(anonymousModelHelper.getNextTupleVarKey(env.enclPkg.packageID));
+
+        BComplexVarSymbol varSymbol = createComplexVarSymbol(varNode.flagSet, varNode.type, varName, env, varNode.pos,
```

Comment:
Can't we just define the symbol here and continue to do the actual `checkTypeAndVarCountConsistency` logic at semantic analysis? I'm not sure if calling `types.checkType` from symbol enter (L1754) is the best approach.

Reply:
Actually, my previous approach is that. used type as 'other' for member variables at the symbol enter. But it causes errors in variable forward referencing.